nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/P2Attributes.java

```diff
@@ -94,9 +94,21 @@ public Builder extension(final String extension) {
     }
 
     public P2Attributes build() {
+      if (isNull(pluginName, componentName, componentVersion, path, fileName, extension, null)) {
+        return null;
+      }
       return new P2Attributes(this);
     }
 
+    private boolean isNull(final String ...args) {
+      for (String a : args) {
+        if (a != null) {
```

Comment:
Question: should it be != or isEmpty?

Reply:
!= is better because it means that value is not assign, and in case isEmpty it could be in property file but empty.